mlrun/model_monitoring/api.py

```diff
@@ -242,6 +223,108 @@ def record_results(
     return model_endpoint
 
 
+def _model_endpoint_validations(
+    model_endpoint: ModelEndpoint,
+    model_path: str = "",
+    sample_set_statistics: typing.Dict[str, typing.Any] = None,
+    drift_threshold: float = None,
+    possible_drift_threshold: float = None,
+):
+    """
+    Validate that provided model endpoint configurations are matching the stored fields of the provided `ModelEndpoint`
+    object. Usually, this method is called by `get_or_create_model_endpoint()` in cases that the model endpoint is
+    already exist. If one of the validations fails, this method might raise an error, indicating on possible conflict.
+
+    :param model_endpoint:           A `ModelEndpoint` object that contains the expected values.
+    :param model_path:               Model store path. Should be similar to the `model_uri` that is stored under
+                                     `model_endpoint.spec.model_uri`. Model endpoint record refers to a single model
+                                     store path.
+    :param sample_set_statistics:    Dictionary of sample set statistics. Once the model endpoint is registered, it
+                                     is forbidden to provide a different reference data to that model endpoint.
+                                     In case of discrepancy between the provided `sample_set_statistics` and the
+                                     `model_endpoints.spec.feature_stats`, a warning will be presented to the user.
+    :param drift_threshold:          The threshold of which to mark drifts. Should be similar to the drift threshold
+                                     that has already assigned to the current model endpoint.
+    :param possible_drift_threshold: The threshold of which to mark possible drifts. Should be similar to the possible
+                                     drift threshold  that has already assigned to the current model endpoint.
+
+    """
+    # Model path
+    if model_path and model_endpoint.spec.model_uri != model_path:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            f"provided model store path {model_path} does not match "
+            f"the path that is stored under the existing model "
+            f"endpoint record: {model_endpoint.spec.model_uri}"
+        )
+
+    # Feature stats
+    if (
+        sample_set_statistics
+        and sample_set_statistics != model_endpoint.status.feature_stats
+    ):
+        logger.warning(
+            "Provided sample set statistics is different from the registered statistics. "
+            "If you wish to use a new statistics as a reference expected data, it is "
+            "recommended to generate a new model endpoint record."
```

Comment:
make the comment more explicit (it's not a recommendation) and non-personal:
Instead of "If you wish.. endpoint record" use: "If new sample set statistics is to be used, new model endpoint should be created"